.github/workflows/ci.yaml

```diff
@@ -86,60 +51,156 @@ jobs:
       #      -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
       #      https://api.github.com/repos/${{ github.actor }}/${{ github.repository }}/actions/runs/${{ github.run_id }}/timing
 
+
+  build_job:
+    timeout-minutes: 30
+    runs-on: ubuntu-20.04
+    needs: test_job
+
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+
+    permissions:      
+      contents: read  #permissions for publishing Docker images to Github Packages
+      packages: write #permissions for publishing Docker images to Github Packages
+    
+    defaults:
+      run:
+        working-directory: angular
+    
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Buld phase
+        run: npm run build
+
+      #---Build Docker image, Login & Publish to Docker Packages---
+      - name: Log in to the Container registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login $REGISTRY -u ${{ github.actor }} --password-stdin
+
+      - name: build Docker image on push event
+        if: github.event_name == 'push'
+        run: docker build -t $REGISTRY/$IMAGE_NAME:${GITHUB_SHA::7} -t $REGISTRY/$IMAGE_NAME:latest .
+
+      - name: build Docker image on PR event
+        if: github.event_name == 'pull_request'
```

Comment:
why do you distinguish between the push and pr events for the build? 
you don't tag it differently - so it just code duplication. same issue with few steps below (push)

Reply:
I've got the wrong(?) idea that commit hash is extracted with different commands on push and PR.
Also official documentation says (https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request):
"Note that GITHUB_SHA for this event is the last merge commit of the pull request merge branch. If you want to get the commit ID for the last commit to the head branch of the pull request, use github.event.pull_request.head.sha instead."
Due to my lack of experience I'm not sure whether I need 'the last merge commit' or 'the last commit to the head branch'.